File: TelerikUIExamples/TelerikUIExamples/CalendarViewModes.m
//
//  CalendarViewModes.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

#import "CalendarViewModes.h"
#import <TelerikUI/TelerikUI.h>

@interface CalendarViewModes () <TKCalendarDelegate>

@property (nonatomic, strong) TKCalendar *calendarView;

@end

@implementation CalendarViewModes

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
        
        [self addOption:@"Week view" selector:@selector(selectWeekView)];
        [self addOption:@"Month" selector:@selector(selectMonth)];
        [self addOption:@"Month Names" selector:@selector(selectMonthNames)];
        [self addOption:@"Year" selector:@selector(selectYear)];
        [self addOption:@"Year Numbers" selector:@selector(selectYearNumbers)];
        [self addOption:@"Flow" selector:@selector(selectFlow)];
        
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.view.backgroundColor = [UIColor lightGrayColor];
    
    NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
    calendar.firstWeekday = 2;
    NSDate *date = [NSDate date];
    NSDateComponents *components = [NSDateComponents new];
    components.year = -10;
    NSDate *minDate = [calendar dateByAddingComponents:components toDate:date options:0];
    components.year = 10;
    NSDate *maxDate = [calendar dateByAddingComponents:components toDate:date options:0];
    
    self.calendarView = [[TKCalendar alloc] initWithFrame:self.view.bounds];
    [self.view addSubview:self.calendarView];

    self.calendarView.delegate = self;
    [self selectYear];
    self.calendarView.calendar = calendar;
    self.calendarView.minDate = minDate;
    self.calendarView.maxDate = maxDate;
// >> view-modes-pinchzoom
    self.calendarView.allowPinchZoom = NO;
// << view-modes-pinchzoom    
    
    [self.calendarView navigateToDate:date animated:NO];
    
// >> view-modes-presenter
    TKCalendarYearPresenter *presenter = (TKCalendarYearPresenter*)self.calendarView.presenter;
    presenter.columns = 3;
// << view-modes-presenter
}

- (void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    if (self.calendarView.viewMode == TKCalendarViewModeWeek) {
        self.calendarView.frame = CGRectMake(self.view.bounds.origin.x, self.view.bounds.origin.y, CGRectGetWidth(self.view.bounds), 100);
    }
    else {
        self.calendarView.frame = self.view.bounds;
    }
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)selectYear
{
// >> getting-started-viewmodeyear
    self.calendarView.viewMode = TKCalendarViewModeYear;
// << getting-started-viewmodeyear    
}

- (void)selectMonth
{
// >> view-modes-month
    self.calendarView.viewMode = TKCalendarViewModeMonth;
// << view-modes-month
}

- (void)selectMonthNames
{
// >> view-modes-monthnames
    self.calendarView.viewMode = TKCalendarViewModeMonthNames;
// << view-modes-monthnames    
}

- (void)selectYearNumbers
{
// >> view-modes-yearnumber
    self.calendarView.viewMode = TKCalendarViewModeYearNumbers;
// << view-modes-yearnumber    
}

- (void)selectFlow
{
// >> view-modes-flow
    self.calendarView.viewMode = TKCalendarViewModeFlow;
// << view-modes-flow    
}

- (void)selectWeekView
{
// >> view-modes-week
    self.calendarView.viewMode = TKCalendarViewModeWeek;
// << view-modes-week
}

#pragma mark TKCalendarDelegate

// >> view-modes-changeviewmode
- (void)calendar:(TKCalendar *)calendar didChangedViewModeFrom:(TKCalendarViewMode)previousViewMode to:(TKCalendarViewMode)viewMode
{
    if (viewMode == TKCalendarViewModeWeek || previousViewMode == TKCalendarViewModeWeek) {
        [self.view setNeedsLayout];
    }
    
    self.selectedOption = (int)viewMode;
}
// << view-modes-changeviewmode



@end

File: TelerikUIExamplesInCSharp/Examples/Calendar/CalendarViewModes.cs
using System;
using System.Drawing;

using Foundation;
using UIKit;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	[Register("CalendarViewModes")]
	public class CalendarViewModes : XamarinExampleViewController
	{
		CalendarDelegate calendarDelegate;

		public TKCalendar CalendarView {
			get;
			set;
		}

		public override void ViewDidLoad ()
		{
			this.AddOption ("Year", SelectYear);
			this.AddOption ("Month", SelectMonth);
			this.AddOption ("Month Names", SelectMonthNames);
			this.AddOption ("Year Numbers", SelectYearNumbers);
			this.AddOption ("Flow", SelectFlow);
			this.AddOption ("Week view", SelectWeekView);

			base.ViewDidLoad ();

			NSCalendar calendar = new NSCalendar (NSCalendarType.Gregorian);
			calendar.FirstWeekDay = 2;
			NSDate date = NSDate.Now;
			NSDateComponents components = new NSDateComponents ();
			components.Year = -10;
			NSDate minDate = calendar.DateByAddingComponents (components, date, (NSCalendarOptions)0);
			components.Year = 10;
			NSDate maxDate = calendar.DateByAddingComponents (components, date, (NSCalendarOptions)0);

			this.CalendarView = new TKCalendar (this.View.Bounds);
			this.CalendarView.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (this.CalendarView);

			this.calendarDelegate = new CalendarDelegate(this);

			this.CalendarView.Delegate = calendarDelegate;
			this.CalendarView.ViewMode = TKCalendarViewMode.Year;
			this.CalendarView.Calendar = calendar;
			this.CalendarView.MinDate = minDate;
			this.CalendarView.MaxDate = maxDate;
			this.CalendarView.NavigateToDate (date, false);
		}

		public override void ViewDidLayoutSubviews ()
		{
			base.ViewDidLayoutSubviews ();
			if (this.CalendarView.ViewMode == TKCalendarViewMode.Week) {
				this.CalendarView.Frame = new CGRect (0, 0, this.View.Bounds.Width, 100);
			} else {
				this.CalendarView.Frame = this.View.Bounds;
			}
		}

		public void SelectYear ()
		{
			this.CalendarView.ViewMode = TKCalendarViewMode.Year;
		}

		public void SelectMonth ()
		{
			this.CalendarView.ViewMode = TKCalendarViewMode.Month;
		}

		public void SelectMonthNames ()
		{
			this.CalendarView.ViewMode = TKCalendarViewMode.MonthNames;
		}

		public void SelectYearNumbers ()
		{
			this.CalendarView.ViewMode = TKCalendarViewMode.YearNumbers;
		}

		public void SelectFlow ()
		{
			this.CalendarView.ViewMode = TKCalendarViewMode.Flow;
		}

		public void SelectWeekView ()
		{
			this.CalendarView.ViewMode = TKCalendarViewMode.Week;
		}

		class CalendarDelegate : TKCalendarDelegate
		{
			CalendarViewModes main;
			public CalendarDelegate (CalendarViewModes main)
			{
				this.main = main;
			}

			public override void DidChangedViewModeFrom (TKCalendar calendar, TKCalendarViewMode previousViewMode, TKCalendarViewMode viewMode)
			{
				if (viewMode == TKCalendarViewMode.Week || previousViewMode == TKCalendarViewMode.Week) {
					this.main.View.SetNeedsLayout ();
				}
			}
		}
	}


}

